### ISSUE TITLE
[HOLD for payment 2023-10-09] [$500] Assign task - App allows to add anything ahead of confirm task URL and maintains the same URL throughout the process

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#27363]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Click on plus and click on assign task
3. Add any title and click Next
4. Click on share somewhere and copy the URL
5. Change the 'share-destination' part of URL to anything and send the modified URL to any chat
6. Open the URL and observe that it opens confirm task page
7. Open any tab, save anything and observe that confirm task URL still has newly added text added in step 5

## Expected Result:
App should display 'Hmm its not here' page or redirect to proper task page if we add any incorrect text ahead of confirm task URL

## Actual Result:
App opens confirm task page and maintains incorrect URL even on saving or editing any field when we add random text ahead of confirm task URL

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.69-0

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/915efc5f-0c05-4cf9-8e4b-89b4d270793c

https://github.com/Expensify/App/assets/115492554/91bb6add-b509-4c92-bef2-847d90834ae0

**Expensify/Expensify Issue URL:**

**Issue reported by:** @Dhanashree-Sawant

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1694015574809879

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016b8f51527c83f586</li>
        <li>Upwork Job ID: 1701974239937085440</li>
        <li>Last Price Increase: 2023-09-20</li>
<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Contributor | 26797065</li>
<li>Dhanashree-Sawant | Reporter | 26797070</li>
</ul></li>
    </ul>
</details>

**Comment by eVoloshchak** [comment]
> Actually now, in the new task flow, the URL is always new/task even we create it in a report. So we should change the route of NewTaskPage to NEW_TASK to make the invalid URL of NewTaskPage will display not found page

This is correct, looking at the code I can't find a case where `reportID` would be included in the URL when creating a new task
@dukenv0307's [proposal](https://github.com/Expensify/App/issues/27363#issuecomment-1717889038) looks good to me!

🎀👀🎀 C+ reviewed!

**Comment by eVoloshchak** [comment]
Not overdue, waiting on @aldo-expensify's review of [this](https://github.com/Expensify/App/issues/27363#issuecomment-1724074740)

**Comment by MitchExpensify** [comment]
Reminder set to pay out on the 9th.

**Comment by MitchExpensify** [comment]
**Payment summary:**

$750 - @eVoloshchak (Eligible for Manual Requests)
$750 - @dukenv0307 requires payment [offer](https://www.upwork.com/nx/wm/offer/26797065) (Contributor)
$250 - @dhanashree-sawant requires payment [offer](https://www.upwork.com/nx/wm/offer/26797070) (Reporter)

**Comment by MitchExpensify** [comment]
Paid @dukenv0307 & @dhanashree-sawant.

Bump on [BZ steps](https://github.com/Expensify/App/issues/27363#issuecomment-1743430694) @eVoloshchak

**Comment by eVoloshchak** [comment]
- The PR that introduced the bug has been identified. Link to the PR: this was the initial implementation in https://github.com/Expensify/App/pull/17322, but according to [this comment](https://github.com/Expensify/App/blob/31e980c4776616cf1fca34addde8ea36e11f98b1/src/pages/NewTaskPage.js#L27) `NOTE: This page is going to be updated in https://github.com/Expensify/App/issues/16855, this is just a placeholder for now`, so I wouldn't call this a regression, rather an edge case that wasn't accounted for initially
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: I don't think an additional discussion is needed, this was just missed during the initial implementation
-  Determine if we should create a regression test for this bug.
This is an edge case that isn't reproducible during normal app usage (you have to manually enter an invalid URL), I think regression test isn't needed here

**Comment by MitchExpensify** [comment]
Thanks! Closing as all that remains is @eVoloshchak's payment which will happen regardless of this being open 👍

**Comment by JmillsExpensify** [comment]
$750 payment approved for @eVoloshchak based on BZ summary.
